app/browser/reducers/bookmarksReducer.js

```diff
@@ -89,6 +94,12 @@ const bookmarksReducer = (state, action, immutableAction) => {
 
         const destinationDetail = bookmarksState.findBookmark(state, action.get('destinationKey'))
         state = syncUtil.updateObjectCache(state, destinationDetail, STATE_SITES.BOOKMARKS)
+
+        // close bookmark bar when going to 0
+        const bookmarkBarItemCount = bookmarksState.getBookmarksWithFolders(state, 0).size
```

Comment:
A little nit-picky... but could we move this code to app/common/lib/bookmarkUtils? Since it's the same as the code above in bookmarkFoldersReducer.js, each of these could be replaced with a single call (maybe the method would be called like `closeBookmarksToolbarIfEmpty` or something like that)

Should make the logic easier to unit test too 😄 👍